gc/gc-iceberg/src/main/java/org/projectnessie/gc/iceberg/DummyProcedure.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.gc.iceberg;
+
+import org.apache.iceberg.spark.procedures.BaseGcProcedure;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+/** Dummy procedure that accepts an input string and returns it. */
+public class DummyProcedure extends BaseGcProcedure {
```

Comment:
why do we need a `DummyProcedure` in the production code? Shouldn't it be in the test code?

Reply:
It will be removed after #3626 , because that PR was too big, I split it into this.

`DummyProcedure` is enough test the framework.

This class cannot be in a test code as the `GcProcedureUtil#loadGcProcedure` source code uses it.
I can still have some reflection way of loading it, but that looks ugly. Hence keeping it in the source.

Reply:
I am not sure if having such a code in the sources of an open-source project won't be confusing for the users. I think we need a 2nd opinion about that, maybe @snazy or @nastra?

Reply:
Yeah, suggestions are welcome. This was the only way I found to split the PR #3626 to be a testable sub PR.
If we cannot have this, then I have to close this PR and review #3626 as a whole PR.